src/autocomplete/shared/extract-statement-positions-from-query.ts

```diff
@@ -38,6 +41,19 @@ export function extractStatementPositionsFromQuery<L extends LexerType, P extend
     parser.addErrorListener(errorListener);
     getParseTree(parser);
 
+    const tokenStatementPositions = extractStatementsUsingTokens(
+        tokenStream,
+        emptySpaceTokens,
+        endStatementToken,
+    );
+    const tokenStatementResult = {
+        statementPositions: tokenStatementPositions,
+        strategy: StatementExtractionStrategy.Tokens,
+    };
+    if (tokenStatementPositions.length > maxAutocompleteIterations) {
```

Comment:
I'd add a comment here, something like "autocomplete extraction is more reliable, but it's too slow on 30+ queries. This is a temprary fix."